exception handling

eloquent, ch.8

(1)
raising an exception jumps out of the current function and its callers
all the way down to the first call that started the current execution
(2)
catching an exception allows to address the problem and then continue to run the program

EXAMPLE
// (1)
function comp(input) {
    if (typeof input === "number") return input + 1;
    throw new Error(input +  " is NaN."); // (1)
}

function input() {
    return comp("3");
}

// (2)
try {
    console.log(input());
} catch (error) { // (2)
    console.log("Something went wrong: " + error.message);
}

